docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1056 @@
+- [Introduction to vectorization with Vector128 and Vector256](#introduction-to-vectorization-with-vector128-and-vector256)
+  * [Code structure](#code-structure)
+    + [Testing](#testing)
+    + [Benchmarking](#benchmarking)
+      - [Custom config](#custom-config)
+      - [Memory alignment](#memory-alignment)
+        * [Enforcing memory alignment](#enforcing-memory-alignment)
+        * [Memory randomization](#memory-randomization)
+  * [Loops](#loops)
+    + [Scalar remainder handling](#scalar-remainder-handling)
+    + [Vectorized remainder handling](#vectorized-remainder-handling)
+    + [Access violation testing](#access-violation-av-testing)
+  * [Loading and storing vectors](#loading-and-storing-vectors)
+    + [Loading](#loading)
+    + [Storing](#storing)
+    + [Casting](#casting)
+  * [Mindset](#mindset)
+    + [Edge cases](#edge-cases)
+    + [Scalar solution](#scalar-solution)
+    + [Vectorized solution](#vectorized-solution)
+  * [Toolchain](#toolchain)
+    + [Creation](#creation)
+    + [Bit operations](#bit-operations)
+    + [Equality](#equality)
+    + [Comparison](#comparison)
+    + [Math](#math)
+    + [Conversion](#conversion)
+    + [Widening and Narrowing](#widening-and-narrowing)
+    + [Shuffle](#shuffle)
+  * [Summary](#summary)
+    + [Best practices](#best-practices)
+
+TL;DR: Go to [Summary](#summary)
+
+# Introduction to vectorization with Vector128 and Vector256
+
+Vectorization is the art of converting an algorithm from operating on a single value at a time to operating on a set of values (vector). It can greatly improve performance at a cost of increased code complexity.
```

Comment:
Perhaps clarifying that it is the number of values operated on "per iteration"?
```suggestion
Vectorization is the art of converting an algorithm from operating on a single value per iteration to operating on a set of values (vector) per iteration. It can greatly improve performance at a cost of increased code complexity.
```